### PR TITLE
Add BucketOptions

### LINKED ISSUE #399: Add BucketOptions for DistributionValue
This is as per Metrics.proto: https://github.com/census-instrumentation/opencensus-proto/blob/master/src/opencensus/proto/metrics/v1/metrics.proto#L209

Counterpart in Java: https://github.com/census-instrumentation/opencensus-java/pull/1484

Implement here : https://github.com/census-instrumentation/opencensus-python/blob/master/opencensus/metrics/export/value.py#L182

### PR DESCRIPTION
**Pull request by c24t**
Replaces `DistributionValue`'s `bucket_bounds` with a new `BucketOptions` class following [the spec](https://github.com/census-instrumentation/opencensus-proto/blob/master/src/opencensus/proto/metrics/v1/metrics.proto#L209-L235). This is a pretty mechanical implementation of the spec, the precondition checks come straight from the spec's comments.

This branch includes changes from #406, I'll rebase once that PR is merged.

Fixes #399.

### CODE REVIEW

File: tests/unit/metrics/export/test_value.py [hunk 1]
```diff
@@ -135,3 +146,22 @@ def test_init(self):
...
+        explicit = value_module.Explicit(bounds)
+        self.assertEqual(explicit.bounds, bounds)
+
+    def test_bad_init(self):
+        with self.assertRaises(ValueError):
+            value_module.Explicit(None)
+        with self.assertRaises(ValueError):
+            value_module.Explicit([])
+        with self.assertRaises(ValueError):
+            value_module.Explicit([0, 1, 2])
```
Comment (mayurkale22) [inline tests/unit/metrics/export/test_value.py]: Could you add a test for negative values?

Comment (c24t) [reply tests/unit/metrics/export/test_value.py]: Added in 85b41d1.
